modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/TestIgnitionManager.java

```diff
@@ -108,33 +142,34 @@ private static InitParameters applyTestDefaultsToClusterConfig(InitParameters pa
                 .metaStorageNodeNames(params.metaStorageNodeNames())
                 .cmgNodeNames(params.cmgNodeNames());
 
+        builder.clusterConfiguration(applyTestDefaultsToConfig(params.clusterConfiguration(), DEFAULT_CLUSTER_CONFIG));
+
+        return builder.build();
+    }
+
+    private static String applyTestDefaultsToConfig(@Nullable String configStr, Map<String, String> defaults) {
+        if (configStr == null) {
+            configStr = "{}";
+        }
+
         ConfigDocument configDocument;
 
-        if (params.clusterConfiguration() == null) {
-            configDocument = ConfigDocumentFactory.parseString("{}");
-        } else {
-            configDocument = ConfigDocumentFactory.parseString(params.clusterConfiguration());
+        try {
+            configDocument = ConfigDocumentFactory.parseString(configStr);
+        } catch (ConfigException e) {
+            // Preserve original broken content, it might be broken on purpose.
```

Comment:
Sorry, what do you mean by "broken on purpose"?

Reply:
Sometimes we test what happens if configuration file is corrupted/contains error. We must preserve the exception